### ISSUE TITLE
return pointer to cell from SetCellValue

### ISSUE THREAD

**Issue by plandem** [360EntSecGroup-Skylar/excelize#105]
To operate low level for own risk. Why do we need it? Sometimes we just need to do some operations directly without all this "higher level logic" to make some changes as fast as possible. So I would suggest to return a pointer to cell at 'SetCellValue'. In that case it will be possible to set style for that cell directly without all this "resolving required cell in map of maps".

e.g.:
```go
cell := xlsx.SetCellValue("sheet1", "A1", "my string")
cell.S := myStyleID
```

I tried this way and got almost no decrease in speed (about 30 second), but with previous way to get same result I need about 9 min.
https://github.com/plandem/excelize/commit/93f949ef2a4156eb57e11652cb34d89b6976f151

https://github.com/360EntSecGroup-Skylar/excelize/issues/60

**Comment by plandem** [comment]
What about higher order struct for cells/rows and etc? Like at tealeg/xlsx?

**Comment by xuri** [comment]
Thanks for your suggestion. File compatibility takes precedence over performance, for this reason, I don't recommend users to operate the exposed file structure directly, so it's necessary to operate the file through the API. In addition, if the number of cells is large and all styles of the cell are same, we can set style with range in a batch instead of set the cell style one by one. I'll continue to focus on performance and continue to do optimization.

**Comment by plandem** [comment]
In my case, I can't use use set style with range. Conditional formatting is not possible too. 

And again....why don't use higher order struct that will provide API to cell/row/sheet? E.g. like at tealeg?


p.s.: what 'File compatibility' are you talking about?

**Comment by plandem** [comment]
Why don't you want to wrap with higher level API like this (just example)?
https://github.com/plandem/excelize/commit/e054ecc88e196a6e62dc86571a13e0508f0c7117

**Comment by plandem** [comment]
here is some benchmarks:
```
BenchmarkOldSetValue-4           2000000              1123 ns/op
BenchmarkOldSetRangeStyle-4         2000            586769 ns/op
BenchmarkOldSetCellStyle-4          2000            575115 ns/op
BenchmarkNewSetValue-4           2000000               923 ns/op
BenchmarkNewSetRangeStyle-4         2000            552119 ns/op
BenchmarkNewSetCellStyle-4       2000000               747 ns/op
```

Old - using current api
New - using higher order API around your api

**Comment by xuri** [comment]
Thanks for your great work. Different from the design of tealeg/xlsx, this lib needsn't the user to handle the ranks of loops and positioning when dealing with discrete cell operations. For example: if we only need to set the value for the cell `C6`, tealeg/xlsx need to loop 6 times to create 6 rows, and then set the cell, but this library only need to specify the coordinates. That's also the reason for can't be merged with tealeg/xlsx (reference issue #2 and #24). In addition, file compatibility is to avoid generating corrupted file and I'll consider whether to refactor the interface to object-oriented style.
